library/src/main/java/software/amazon/location/auth/AuthHelper.kt

```diff
@@ -15,44 +15,62 @@ class AuthHelper(private val context: Context) {
      * @param identityPoolId The identity pool id for authentication.
      * @return A LocationCredentialsProvider object.
      */
-    fun authenticateWithCognitoIdentityPool(
+    suspend fun authenticateWithCognitoIdentityPool(
         identityPoolId: String,
-    ): LocationCredentialsProvider = LocationCredentialsProvider(
-        context,
-        identityPoolId,
-        // Get the region from the identity pool id
-        Regions.fromName(identityPoolId.split(":")[0]),
-    )
+    ): LocationCredentialsProvider {
+        return withContext(Dispatchers.Default) {
```

Comment:
if you make network calls, do not use Dispatchers.Default, use Dispatchers.IO

Reply:
Done